firewall.go

```diff
@@ -89,6 +131,12 @@ func (s *Session) Close() error {
 	if s.handle == 0 {
 		return nil
 	}
+
+	// if we have a transaction in progress that was not commited, abort it now
+	if s.status.State == BeganTransaction {
+		s.AbortTransaction()
+	}
+
```

Comment:
@rob-no We will not automatically commit a transaction.  That is up to the caller to do.  However, if we detect we have began a transaction without committing it, on Close, we will abort the transaction.

Reply:
The 4 functions in WFP that use transactions (Rules, SetRules, DeleteAll, and SaveRulesToDisk, all start their sessions in a transaction and call commit (unless given an override through the configuration task).
Here is an example of a commit being used:
https://github.com/huntresslabs/agent/blob/dbb8bcfe728b7a375f15852daa838c9f1c584b14/internal/hlib/wfp/wfp_windows.go#L54

Reply:
Ok I see that now. Feels a bit weird that the `beginTransaction` is abstracted away, but the commit isn't.

I think it's fine in this case, but in the future we shouldn't mix the pattern like that IMO

Reply:
Just for additional info, it's a startup option to begin a transaction when the session is created.  You don't have to start a transaction at that point, but if you don't you'll miss some of the operations in the session create function.  You can tell it to start a session without starting a transaction and then call BeginTransaction (but as mentioned, then some of the WFP calls are not going to be a part of the transaction)